src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4117,213 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
+                }
+                else
+                {
+                    assert(jmp == emitJumpList);
+                    emitJumpList = jmp->idjNext;
+                }
+
+                UNATIVE_OFFSET codeSize = jmp->idCodeSize();
+                jmp->idCodeSize(0);
+
+                // setup the next iteration
+                {
+                    instrDescJmp* nextJmp = jmp->idjNext;
+                    jmp->idjNext          = nullptr;
+                    jmp                   = nextJmp;
+                }
+
+                // do not use jmp after this point because it is now the next jmp
+
+                previousTruncatedGroup = jmpGroup;
+
+                jmpGroup->igSize -= (unsigned short)codeSize;
+                jmpGroup->igFlags |= IGF_UPD_ISZ;
+
+                emitTotalCodeSize -= codeSize;
+                totalRemovedSize += codeSize;
+                totalRemovedCount += 1;
+
+                // go to the loop condition, we have setup the next jmp while editing the list
+                continue;
+            }
+#if DEBUG
+            else
+            {
+                if (targetGroup == nullptr)
+                {
+                    JITDUMP("IN%04x from the last instruction in IG%02u jump target is not set!, keeping.\n",
+                            jmp->idDebugOnlyInfo()->idNum, jmpGroup->igNum);
+                }
+                else if ((jmpGroup->igFlags & IGF_HAS_ALIGN) != 0)
+                {
+                    JITDUMP("IN%04x from the last instruction in IG%02u contains alignment, keeping.\n",
+                            jmp->idDebugOnlyInfo()->idNum, jmpGroup->igNum);
+                }
+                else if (jmpGroup->igNext != targetGroup)
+                {
+                    JITDUMP("IN%04x from the last instruction in IG%02u to does not jump to the next group, keeping.\n",
+                            jmp->idDebugOnlyInfo()->idNum, jmpGroup->igNum);
+                }
+            }
+#endif
+        }
+
+        previousJmp = jmp;
+        jmp         = jmp->idjNext;
+    }
+
+    jmp = nullptr;
+
+    // make sure that any instruction groups following the last truncated group have their offset adjusted
+    if (previousTruncatedGroup != nullptr)
+    {
+        for (insGroup* offsetGroup = previousTruncatedGroup->igNext; offsetGroup != nullptr;
+             offsetGroup           = offsetGroup->igNext)
+        {
+            JITDUMP("Adjusted offset of " FMT_BB " from %04X to %04X\n", offsetGroup->igNum, offsetGroup->igOffs,
+                    offsetGroup->igOffs - totalRemovedSize);
+            offsetGroup->igOffs -= totalRemovedSize;
+            assert(IsCodeAligned(offsetGroup->igOffs));
+        }
+    }
+
+#ifdef DEBUG
+    if (EMIT_INSTLIST_VERBOSE)
+    {
+        printf("\nInstruction group list after unconditional jump to next instruction removal:\n\n");
+        emitDispIGlist(false);
+    }
+    if (EMITVERBOSE)
+    {
+        emitDispJumpList();
+    }
```

Comment:
Only print all this if something was removed? Else, print "No unconditional jump removed"?